### PR TITLE
refactor: Convert to jpm addon.

### PR DESCRIPTION
**Pull request by Popog**
Remove custom install.rdf and bootstrap.js, use the generated ones instead.
Move necessary startup code into main.js.
Remove options.xul, use package.json preferences instead.
Convert verticaltabs.jsm to verticaltabs.js to simplify constructor.
Refactor utils.js to remove any trace of Home Dash Utility.

### CODE REVIEW

**Comment by bwinton** [comment]
Ooh, exciting…  Can you rebase this on the latest master and fix the conflicts?

File: utils.js [hunk 1]
```diff
@@ -11,15 +11,14 @@
...
  *
- * The Initial Developer of the Original Code is The Mozilla Foundation.
+ * The Initial Developer of the Original Code is
+ * Philipp von Weitershausen.
  * Portions created by the Initial Developer are Copyright (C) 2011
  * the Initial Developer. All Rights Reserved.
  *
  * Contributor(s):
- *   Edward Lee <edilee@mozilla.com>
- *   Erik Vold <erikvvold@gmail.com>
```
Comment (bwinton) [inline utils.js]: Changing the attribution and contributors of this code seems like an odd thing to do…

File: utils.js [hunk 2]
```diff
@@ -35,126 +34,22 @@
  *
  * ***** END LICENSE BLOCK ***** */
 
-/* global Cc:false, Ci:false */
-/*exported sendPing, addPingStats, watchWindows*/
-'use strict';
```
Comment (bwinton) [inline utils.js]: We definitely want the `'use strict';` in there.

File: utils.js [hunk 3]
```diff
@@ -35,126 +34,22 @@
...
-  }
-  Services.ww.registerNotification(windowWatcher);
+/*
+ d8888b.  .d8b.  db    db db       .d88b.   .d8b.  d8888b.
+ 88  `8D d8' `8b `8b  d8' 88      .8P  Y8. d8' `8b 88  `8D
+ 88oodD' 88ooo88  `8bd8'  88      88    88 88ooo88 88   88
+ 88~~~   88~~~88    88    88      88    88 88~~~88 88   88
+ 88      88   88    88    88booo. `8b  d8' 88   88 88  .8D
+ 88      YP   YP    YP    Y88888P  `Y88P'  YP   YP Y8888D'
+*/
```
Comment (bwinton) [inline utils.js]: I'm not a fan of the huge figlet headers.  Please remove them.  (If you want to add `/* PAYLOAD */` instead that would be fine by me…  🙂)

File: utils.js (hunk 1 above)
Comment (Popog) [inline utils.js]: Those contributors were for the Home Dash Utility (unload / watchers) which no longer exist. This utils file was original a stripped down version of https://github.com/mozilla/prospector/blob/master/homeDash/scripts/utils.js

The real reason is actually that I just deleted this file as a part of my refactor and then later remade it and copy pasted a header :P. I can revert this if you want.

File: utils.js (hunk 2 above)
Comment (Popog) [inline utils.js]: Whoops.

**Comment by bwinton** [comment]
When I run it with `jpm run`, I get an `ReferenceError: Event is not defined` at verticaltabs.js line 538, and `TypeError: stats is undefined` at utils.js line 70…  I think they might be related to the switch from jsm to js, or the different context of Add-on SDK code vs. Bootstraped XUL code.

**Comment by Popog** [comment]
Fixed Event reference error as well as another error related to stats (also some style changes for parameterless constructors).

Complaints appreciated  🐶.

**Comment by bwinton** [comment]
Yeah, please roll back the header…  :)
I also found a `TypeError: Not enough arguments to Event.initEvent.` from the call to `emit` at verticaltabs.js line 539, which is causing the urls not to show.  (And maybe other stuff…)

**Comment by Popog** [comment]
Wow, I am just terrible at debugging today.
Really really fixed this time 🤞. Also header rolled back on utils.js.

**Comment by bwinton** [comment]
I just noticed that we aren't calling the `sendPing` method anymore.  This is going to be a problem when we try to get the Telemetry data from the back-end…  😉  It should be called every 24 hours and on shutdown, as per https://github.com/bwinton/TabCenter/pull/482/files#diff-b83c0bde60e252508b3088b391d3c14fL124  (It looks like you can just add it to the bottom of the `sendPayload` method, for a quick fix.)

**Comment by Popog** [comment]
Derp... that is exactly where it was supposed to go. Fixed.

Maybe I need more sleep or something.

**Comment by bwinton** [comment]
Hey Popog, I think this is good enough to merge now.  Would you mind getting the latest copy of master, and squashing your commits down to a single one so that I can merge it?  Thanks!

**Comment by Popog** [comment]
Tried. Only ended up adding more commits. Can you do this https://help.github.com/articles/about-pull-request-merge-squashing/

**Comment by bwinton** [comment]
Done and done.  Thank you!  😄

**Pull request merged**